services/wireguard/service.go

```diff
@@ -49,13 +58,15 @@ type Manager struct {
 	locationResolver location.Resolver
 	ipResolver       ip.Resolver
 	isStarted        bool
+	wgClient         *wireguardctrl.Client
 }
 
 // Config represent a Wireguard service provider configuration that will be passed to the consumer for establishing a connection
 type Config struct {
 	PublicKey string
-	IP        string
 	Endpoint  string
+	PeerKey   string // TODO peer private key should be generated on consumer side
+	PeerIP    string
```

Comment:
I would like to have more indication which part of config is related to provider and which part to consumer.
I.E. PublicKey -> ProviderPublicKey
Endpoint -> ProviderEndpoint
PeerKey -> ConsumerPrivateKey
PeerIP -> ConsumerIP

Reply:
what if we will make nested JSON?

```JSON
{
	"Provider": {
		"PublicKey": "aaaaaaaaaaaa",
		"Endpoint": "1.2.3.4:12345"
	},
	"Consumer": {
		"PrvivateKey": "bbbbbbbbbbb",
		"IP": "192.168.1.1"
	}
}
```

Reply:
Looks better 👍